src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -946,15 +945,10 @@ public Consumer createConsumerFromDTO(ConsumerDTO consumer, ConsumerType type, P
         // API:registerConsumer
         Set<String> keyStrings = splitKeys(activationKeys);
 
-        // Only let NoAuth principals through if there are activation keys to consider:
-        if ((principal instanceof NoAuthPrincipal) && keyStrings.isEmpty()) {
-            throw new ForbiddenException(i18n.tr("Insufficient permissions"));
-        }
-
         validateOnKeyStrings(keyStrings, owner.getKey(), userName);
 
         // Raise an exception if none of the keys specified exist for this owner.
-        List<ActivationKey> keys = checkActivationKeys(principal, owner, keyStrings);
+        List<ActivationKey> keys = this.activationKeyCurator.findByKeyNames(owner.getKey(), keyStrings);
```

Comment:
We should at least retain the old behavior of logging any keys that aren't found during this lookup.

Reply:
Done.